### PR TITLE
JP-3079 Ensure all imprint exposures are include in associations

### PR DESCRIPTION
**Pull request by stscieisenhamer**
Resolves [JP-3079](https://jira.stsci.edu/browse/JP-3079)

This PR addresses the issue that not all imprint exposures that should be associated with science exposures were being associated. The bug was that imprints were not being reprocessed so that they can be matched against associations created after the imprints were processed.

**Checklist for maintainers**
- [x] added entry in `CHANGES.rst` within the relevant release section
- [x] updated or added relevant tests
- [x] updated relevant documentation
- [x] added relevant milestone
- [x] added relevant label(s)
- [x] ran regression tests, post a link to the Jenkins job below.
      [How to run regression tests on a PR](https://github.com/spacetelescope/jwst/wiki/Running-Regression-Tests-Against-PR-Branches)
- [x] Make sure the JIRA ticket is [resolved properly](https://github.com/spacetelescope/jwst/wiki/How-to-resolve-JIRA-issues)

### CODE REVIEW

**Review by jemorrison — APPROVED** [review]
The code seems reasonable - I have a bit of trouble understanding association code. But the imprints are taking higher importance than background . The rest I am going to trust you got it correct.

**Comment by jemorrison** [comment]
missing change log

**Review by kmacdonald-stsci — APPROVED** [review]

**Comment by stscieisenhamer** [comment]
[Regression test](https://plwishmaster.stsci.edu:8081/job/RT/job/eisenhamer-dev/301/)

**Comment by hbushouse** [comment]
> [Regression test](https://plwishmaster.stsci.edu:8081/job/RT/job/eisenhamer-dev/301/)

So ... are all the failures expected and OK?  I'm surprised that changing the rules for imprints would affect so many different tests, because imprints are limited to NIRSpec.

File: CHANGES.rst [hunk 1]
```diff
@@ -1,6 +1,11 @@
 1.9.4 (unreleased)
 ==================
 
+associations
+------------
+
+- Ensure all imprint exposures are include in associations [#7438]
```
Comment (hbushouse) [inline CHANGES.rst:7]: minor nit: typo  "include" should  be "included"

Comment (hbushouse) [reply CHANGES.rst:7]: But kudos for correct usage of "ensure"  (as opposed to "insure")!

**Comment by stscieisenhamer** [comment]
[Regression test try #2](https://plwishmaster.stsci.edu:8081/job/RT/job/eisenhamer-dev/302/).

First one had a bad artifactory specification. Thanks for the heads-up regressions...

**Comment by hbushouse** [comment]
> [Regression test try #2](https://plwishmaster.stsci.edu:8081/job/RT/job/eisenhamer-dev/302/).
> 
> First one had a bad artifactory specification. Thanks for the heads-up regressions...

Woot. The test run is clean.

So I'm going to merge and leave the minor nit pick in the change log to be fixed later.

**Review by hbushouse — APPROVED** [review]

**Pull request merged**